crates/recursor/src/recursor_pool.rs

```diff
@@ -46,14 +48,27 @@ pub(crate) struct RecursorPool<P: ConnectionProvider> {
     zone: Name,
     ns: NameServerPool<P>,
     active_requests: Arc<Mutex<HashMap<Query, SharedLookup>>>,
+    #[cfg(feature = "metrics")]
+    iterative_query_counter: Counter,
 }
 
 impl<P: ConnectionProvider> RecursorPool<P> {
     pub(crate) fn from(zone: Name, ns: NameServerPool<P>) -> Self {
+        #[cfg(feature = "metrics")]
+        let iterative_query_counter = counter!("hickory_iterative_queries_total");
```

Comment:
Do you think the concept of "iterative query" is self-explanatory? I don't find it all that obvious, but then I am less familiar in practice with recursing DNS servers etc.

Reply:
I'm going to change this to `hickory_recursor_outgoing_queries_total`. That has the nice benefit that multiple metrics will share the `hickory_recursor_` prefix.

FYI I'm going to redo the tests to avoid internet access before force-pushing.